### ISSUE TITLE
`npm run test --output` doesn't work on Android.

### ISSUE THREAD

**Issue by wknapik** [brave/brave-browser#15595]
## Description 
On all platforms, the `--output` switch can be passed to `npm run test`, to create an xunit test report file, which can be parsed by Jenkins. This doesn't work on iOS, or Android. On iOS, there's a simple workaround - the stdout can be piped to `xcpretty` and you get the report you need. But on Android, there is no good workaround.

## Steps to reproduce 

   1. `npm run test -- brave_unit_tests Release --target_os=android --target_arch=x86 --output=file.name`
   2. The xunit report file is not created.

## Actual result 
No report.

## Expected result
Yes report.

## Issue reproduces how often 
Easily reproduced

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? Yes
- Can you reproduce this issue with the current Play Store Beta version? Yes
- Can you reproduce this issue with the current Play Store Nightly version? Yes

### Additional information
As a workaround, I'm trying to use `--json-results-file`, which produces a report in a custom json format, but there's multiple problems with that. The format is not recognized by any Jenkins plugin (as opposed to xunit, which is native to Jenkins) and the report file is also overwritten halfway through execution, because the npm script injects another test suite that was not requested on the command line. So the file is created for the first test suite and them immediately overwritten by the next one.

**Comment by bridiver** [comment]
This is correct, there is no `output` or `gtest_output` option for the android test runner. I have a workaround for this here https://github.com/brave/brave-core/pull/29870/files
